battenberg/utils.py

```diff
@@ -1,19 +1,31 @@
 import os
-from pygit2 import Repository, discover_repository, init_repository, Keypair
+import logging
+import re
+import subprocess
+from typing import Optional
+from pygit2 import discover_repository, init_repository, Keypair, Repository
 
 
-def open_repository(path):
+logger = logging.getLogger(__name__)
+
+
+def open_repository(path: str) -> Repository:
     return Repository(discover_repository(path))
 
 
-def open_or_init_repository(path):
+def open_or_init_repository(path: str, template: str, initial_branch: Optional[str] = None):
     try:
         return open_repository(path)
     except Exception:
         # Not found any repo, let's make one.
         pass
 
-    repo = init_repository(path)
+    repo = init_repository(path, initial_head=initial_branch)
+
+    # Mirror the default HEAD of the template repo if client hasn't explicitly provided it.
+    if not initial_branch:
+        repo = set_initial_branch(repo, template)
```

Comment:
since `repo` is pass-by-ref here, the target is already set on L51, making it not necessary to reassign `repo` here